plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -937,11 +938,18 @@ def create_vm(module, proxmox, vmid, newid, node, name, memory, cpu, cores, sock
             kwargs['searchdomain'] = ' '.join(searchdomains)
 
     # VM tags are expected to be valid and presented as a comma/semi-colon delimited string
-    if module.params['tags']:
-        for tag in module.params['tags']:
-            if not re.match(r'^[a-z0-9_][a-z0-9_\-\+\.]*$', tag):
-                module.fail_json(msg='%s is not a valid tag' % tag)
-        kwargs['tags'] = ",".join(module.params['tags'])
+    if 'tags' in kwargs:
+        # Set tags to null if empty list
+        if not kwargs['tags']:
+            kwargs['tags'] = ""
+        # Do nothing if tags contains a single empty element
+        elif len(kwargs['tags']) == 1 and not kwargs['tags'][0]:
```

Comment:
I'm not sure what this case should be. If you don't want `tags`, simply don't specify it or write `tags: ~` in the task.

If you want to specify tags, either specify `tags: []` for an empty list, or a non-empty list of valid tags.